pkg/support/dump_test.go

```diff
@@ -15,21 +15,129 @@
 package support
 
 import (
+	"fmt"
+	"path/filepath"
+	"strings"
 	"testing"
+	"time"
 
+	"github.com/spf13/afero"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/utils/exec"
+	exectesting "k8s.io/utils/exec/testing"
+
+	"antrea.io/antrea/pkg/util/runtime"
 )
 
-func TestParseLogDate(t *testing.T) {
-	data := "I0817 06:55:10.804384       1 shared_informer.go:270] caches populated"
-	ts, err := parseTimeFromLogLine(data, "2021", "antrea-agent")
-	assert.Nil(t, err)
-	assert.Equal(t, ts.String(), "2021-08-17 06:55:10 +0000 UTC")
+type testExecuter struct {
+	exectesting.FakeExec
+}
 
-	data = "2021-06-01T09:30:43.823Z|00004|memory|INFO|cells:299 monitors:2 sessions:2"
-	ts, err = parseTimeFromLogLine(data, "2021", "ovs")
-	assert.Nil(t, err)
-	assert.Equal(t, ts.String(), "2021-06-01 09:30:43 +0000 UTC")
+// Command func is to track correct commands are executed
+func (te *testExecuter) Command(cmd string, args ...string) exec.Cmd {
+	fakeCmd := new(exectesting.FakeCmd)
+	fakeCmd.CombinedOutputScript = append(fakeCmd.CombinedOutputScript, func() ([]byte, []byte, error) {
+		return []byte(fmt.Sprintf("%s %s", cmd, strings.Join(args, " "))), nil, nil
+	})
+	return fakeCmd
+}
+
+func TestDumpLog(t *testing.T) {
+	basedir := "dir1/dir2"
+	fs := afero.NewMemMapFs()
+	dumper := NewAgentDumper(fs, nil, nil, nil, nil, "7s", true, true)
+	err := dumper.DumpLog(basedir)
+	var x []byte
+	if err != nil {
+		assert.Error(t, err)
+	} else {
+		if runtime.IsWindowsPlatform() {
+			x, err = afero.ReadFile(fs, "dir1/dir2/logs/agent")
+			if err == nil {
+				if string(x) == "rancher-wins-antrea-agent" {
+					assert.NoError(t, err)
+				}
+			}
+			x, err = afero.ReadFile(fs, "dir1/dir2/logs/ovs")
+			if err == nil {
+				if string(x) == "ovs" {
+					assert.NoError(t, err)
+				}
+			}
+			x, err = afero.ReadFile(fs, "dir1/dir2/logs/kubelet")
+			if err == nil {
+				if string(x) == "kubelet" {
+					assert.NoError(t, err)
+				}
+			}
+		} else {
```

Comment:
IMO, this test is not written in a readable way. I am not even sure what you are testing. For example:
```go
			x, err = afero.ReadFile(fs, "dir1/dir2/logs/kubelet")
			if err == nil {
				if string(x) == "kubelet" {
					assert.NoError(t, err)
				}
			}
```
Isn't `dir1/dir2/logs/kubelet` supposed to be a directory? Why would you get `"kubelet"` as file contents when reading `dir1/dir2/logs/kubelet`?

IMO you should populate some directories & files in the correct locations in `fs` before calling `DumpLog`, then check that `DumpLog` behaves as expected.